include/cudaq/Optimizer/Dialect/CC/CCTypes.h

```diff
@@ -1,14 +1,14 @@
-/****************************************************************-*- C++ -*-****
+/*************************************************************** -*- C++ -*- ***
  * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
  * All rights reserved.                                                        *
  *                                                                             *
  * This source code and the accompanying materials are made available under    *
  * the terms of the Apache License 2.0 which accompanies this distribution.    *
- ******************************************************************************/
+ *******************************************************************************/
 
 #pragma once
 
-#include "llvm/ADT/SmallVector.h"
+#include "cudaq/Optimizer/Dialect/Common/SmallVector.h"
```

Comment:
@boschmitt and @schweitzpgi might comment here - is this inline with LLVM style to create our own `SmallVector.h` that filters out LLVM warnings? My though was that we'd want to name it something more explicit (indicating it is not a re-implementation or simple forward declarations), e.g. `SmallVectorWrapper.h`.

Reply:
I think including llvm/ADT/SmallVector.h by default is the correct thing to do - warnings should not be disabled by default but only if necessary and after checking each one. I.e. if you can use llvm/ADT/SmallVector.h without getting a warning you should. If you get a warning, a quick search of the code base for that include makes sense to determine if it makes sense to create or use a separate header file to disable the same warning in multiple places.